kcidb/test_oo.py

```diff
@@ -203,3 +204,225 @@ def test_two_object(self):
             '>checkout#>build#>test#',
             {"test", "build", "checkout", "revision"}, 2
         )
+
+
+class KCIDBTraversingTestCase(kcidb.unittest.TestCase):
+    """Test case for traversing objects."""
+
+    # Conform to unittest conventions, pylint: disable=invalid-name
+
+    def query_str(self, pattern_string):
+        """Run OO client query with a parsed string."""
+        return self.client.query(
+            kcidb.orm.Pattern.parse(pattern_string)
+        )
+
+    def assertNumberOfInstances(self, oo_object_list, num_instances):
+        """
+        Check that the correct number of instances is returned for
+        the object list.
+        Args:
+            oo_object_list: A list of kcidb.oo.Object instances.
+            num_instances:  The number of instance of object contained
+                            in the oo_object_list.
+        """
+        assert isinstance(oo_object_list, list)
+        assert isinstance(num_instances, int)
+
+        self.assertEqual(
+            len(oo_object_list), num_instances,
+            msg=f"Invalid number of {oo_object_list[0]} instances"
```

Comment:
This would fail with another exception if the `oo_object_list` is empty, masking the assertion failure. I.e. you can't really find out the expected type of objects in an empty list. Additionally, strictly speaking, the type of one object doesn't guarantee that the rest of them are the same type. To make this work reliably and make sense you'd need to pass the (name of the) expected type to the function explicitly.